Resources/Storage/StorageElement.py

```diff
@@ -397,39 +395,75 @@ def getOccupancy(self, unit='MB', **kwargs):
     filteredPlugins = self.__filterPlugins('getOccupancy')
     if not filteredPlugins:
       return S_ERROR(errno.EPROTONOSUPPORT, "No storage plugins to query the occupancy")
+
+    # Call occupancy plugin if requested
+    occupancyPlugin = self.options.get('OccupancyPlugin')
+    if occupancyPlugin:
+      res = ObjectLoader().loadObject(occupancyPlugin)
+      if not res['OK']:
+        return S_ERROR(errno.EPROTONOSUPPORT, 'Failed to load occupancy plugin %s' % occupancyPlugin)
+      log.verbose('Use occupancy plugin %s' % occupancyPlugin)
+      try:
+        occupancyPlugin = res['Value'](self)
+        res = occupancyPlugin.getOccupancy(**kwargs)
```

Comment:
I think in case of error here, or in case of exception, you should return it, and not continue. If a occupancy plugin is defined in the config, then that's what we want to use for sure

Reply:
Sure. The code has been changed...